java/arcs/core/entity/StorageAdapter.kt

```diff
@@ -63,12 +64,12 @@ class EntityStorageAdapter<T : Entity>(
     private val ttl: Ttl,
     private val time: Time,
     private val dereferencerFactory: EntityDereferencerFactory,
-    private val storageKey: StorageKey
+    private val storageKey: StorageKey,
+    private val restrictedEntitySchema: Schema? = null
```

Comment:
When do we know the restriction information?

Is it possible we could update the API of `EntityBase` so it accepts a serialization/restriction Schema? And then we don't need to plumb it in here?

Reply:
(Then at code generation time, the concrete schema can pass the appropriate restriction schema to the superclass). 

I am making the assumption that we will always want to write this particular type with restriction (i.e., the restriction is inherently part of the generated type), not sure if this assumption is true or not.

Reply:
the restricted schema is available at build time.
refactored the code to pass the restricted schema to EntityBase in ctor.

Reply:
oh damn, I seem to be arguing for the opposite

Reply:
> I am making the assumption that we will always want to write this particular type with restriction (i.e., the restriction is inherently part of the generated type), not sure if this assumption is true or not.

I think we should not be making this assumption - this would essentially tie a particle to one and only recipe, which is very strongly against the Arcs computation model. There are also cases where this assumption is already wrong - in FedEx redactors, which are first generic particles that will be used in many different recipes.

Reply:
This is replaying the conversation with had a BAR a few weeks ago - whether to perform field dropping inside the particle with codegen, or inside the handle. The resolution was to perform it inside a handle based on the handle type.

Reply:
Got it, I guess I should have reviewed the notes from that discussion first. :-(

I thought that we generate a new type per *handle*, but it seems that we do it per *particle*. 

I wonder if it makes sense to generate a subclass of each entity per handle that holds the serializationSchema for the handle? This might lead to some crazy type wrangling that we don't want, though.

Anyway, sorry for leading you astray Maria, I guess based on this discussion, the original serialize(schema) approach is needed. 

I still can't resist the feeling that we should be encoding this at the type level (i.e., creating a type per handle), but I think I am going to be able to slowly convince myself otherwise with some more thinking.

Reply:
thanks Piotr, updated the code back to first commit like.